ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,829 @@
+
+/**
+ * @param mysqli_result $mysql_result
+ * @param int $offset
+ *
+ * @return object|bool
+ */
+function mysqli_fetch_field_direct(mysqli_result $mysql_result, int $offset) {}
+
+/**
+ * @param mysqli_result $mysql_result
+ *
+ * @return array|bool
+ */
+function mysqli_fetch_lengths(mysqli_result $mysql_result) {}
+
+/**
+ * @param mysqli_result $mysql_result
+ * @param int $mode
+ *
+ * @return array|bool
+ */
+function mysqli_fetch_all(mysqli_result $mysql_result, int $mode = MYSQLI_NUM) {}
+
+/**
+ * @param mysqli_result $mysql_result
+ * @param int $fetchtype
+ *
+ * @return array|bool
+ */
+function mysqli_fetch_array(mysqli_result $mysql_result, int $fetchtype = MYSQLI_BOTH) {}
+
+function mysqli_fetch_assoc(mysqli_result $mysql_result): array {}
+
+function mysqli_fetch_object(mysqli_result $mysqli_result, string $class_name = 'stdClass', array $params = []): object {}
+
+function mysqli_fetch_row(mysqli_result $mysqli_result): array {}
+
+function mysqli_field_count(mysqli_result $mysqli_result): int {}
+
+function mysqli_field_seek(mysqli_result $mysqli_result, int $field_nr): bool {}
+
+function mysqli_field_tell(mysqli_result $mysqli_result): int {}
+
+function mysqli_free_result(mysqli_result $mysqli_result): void {}
+
+function mysqli_get_connection_stats(mysqli $mysqli_link): array {}
+
+function mysqli_get_client_stats(mysqli $mysqli_link): array {}
+
+function mysqli_get_charset(mysqli $mysqli_link): object {}
+
+function mysqli_get_client_info(): string {}
+
+function mysqli_get_client_version(): int {}
+
+function mysqli_get_links_stats(mysqli $mysqli_link): array {}
+
+function mysqli_get_host_info(mysqli $mysqli_link): string {}
+
+function mysqli_get_proto_info(mysqli $mysqli_link): int {}
+
+function mysqli_get_server_info(mysqli $mysqli_link): string {}
+
+function mysqli_get_server_version(mysqli $mysqli_link): int {}
+
+function mysqli_get_warnings(mysqli $mysqli_link): mysqli_warning {}
+
+function mysqli_init(): mysqli {}
+
+function mysqli_info(mysqli $mysqli_link): string {}
+
+/**
+ * @param mysqli $mysqli_link
+ *
+ * @return int|string
+ */
+function mysqli_insert_id(mysqli $mysqli_link) {}
+
+function mysqli_kill(mysqli $mysqli_link, int $processid): bool {}
+
+function mysqli_more_results(mysqli $mysqli_link): bool {}
+
+function mysqli_multi_query(mysqli $mysqli_link, string $query): bool {}
+
+function mysqli_next_result(mysqli $mysqli_link): bool {}
+
+function mysqli_num_fields(mysqli $mysqli_link): int {}
+
+/**
+ * @param mysqli $mysqli_link
+ *
+ * @return int|string
+ */
+function mysqli_num_rows(mysqli $mysqli_link) {}
+
+function mysqli_options(mysqli $mysqli_link, int $mysql_option, mixed $mysql_value): bool {}
+
+function mysqli_ping(mysqli $mysqli_link): bool {}
+
+function mysqli_poll(?array &$read, ?array &$write, ?array &$error, int $sec, int $usec = 0): int {}
+
+/**
+ * @param mysqli $mysqli_link
+ * @param string $query
+ *
+ * @return mysqli_stmt|bool
```

Comment:
I've just committed https://github.com/php/php-src/commit/0cec268d15e5aded8692ee8076e93d2839725918 to add support for directly using `: mysqli_stmt|bool`.

Reply:
when I try to use this diff:
```
-/**
- * @param mysqli $mysql_link
- *
- * @return mysqli_result|bool
- */
-function mysqli_use_result(mysqli $mysql_link) {}
+function mysqli_use_result(mysqli $mysql_link): mysqli_result|false {}
```
I get:
```
$ php scripts/dev/gen_stub.php ext/mysqli/mysqli.stub.php
In ext/mysqli/mysqli.stub.php:
Unimplemented
```

Reply:
Did you rebase the PR onto current master? Then it should work and yield:
````.c
ZEND_BEGIN_ARG_WITH_RETURN_OBJ_TYPE_MASK_EX(arginfo_mysqli_use_result, 0, 1, mysqli_result, MAY_BE_FALSE)
	ZEND_ARG_OBJ_INFO(0, mysql_link, mysqli, 0)
ZEND_END_ARG_INFO()
````

Reply:
check!